test/contacts.js

```diff
@@ -78,172 +156,232 @@ describe('contacts', function() {
   })
 
   describe('getByEmail', function() {
-    it('should return a contact record based on the email', function() {
-      return hubspot.contacts
-        .getByEmail('testingapis@hubspot.com')
-        .then(data => {
-          expect(data).to.be.a('object')
-          expect(data.properties).to.be.a('object')
+    let email = 'testingapis@hubspot.com'
+    const contactByEmailEndpoint = {
+      path: `/contacts/v1/contact/email/${email}/profile`,
+      response: { properties: {} },
+    }
+    fakeHubspotApi.setupServer({ getEndpoints: [contactByEmailEndpoint] })
+
+    before(function() {
+      if (process.env.NOCK_OFF) {
+        return hubspot.contacts.get({ count: 1 }).then(data => {
+          email = emailsFromContacts(data.contacts)[0]
         })
+      }
+    })
+
+    it('should return a contact record based on the email', function() {
+      return hubspot.contacts.getByEmail(email).then(data => {
+        expect(data).to.be.a('object')
+        expect(data.properties).to.be.a('object')
+      })
     })
   })
 
   describe('getByEmailBatch', function() {
-    it('should return a contact record based on a array of emails', function() {
-      return hubspot.contacts
-        .getByEmailBatch([
-          'testingapis@hubspot.com',
-          'testingapisawesomeandstuff@hubspot.com',
-        ])
-        .then(data => {
-          expect(data).to.be.an('object')
+    let emails = [
+      'testingapis@hubspot.com',
+      'testingapisawesomeandstuff@hubspot.com',
+    ]
+    const contactByEmailsEndpoint = {
+      path: '/contacts/v1/contact/emails/batch',
+      response: { properties: {} },
+      query: { email: emails },
+    }
+    fakeHubspotApi.setupServer({ getEndpoints: [contactByEmailsEndpoint] })
+
+    before(function() {
+      if (process.env.NOCK_OFF) {
+        return hubspot.contacts.get({ count: 3 }).then(data => {
+          emails = emailsFromContacts(data.contacts)
         })
+      }
+    })
+
+    it('should return a contact record based on a array of emails', function() {
+      return hubspot.contacts.getByEmailBatch(emails).then(data => {
+        expect(data).to.be.an('object')
+      })
     })
   })
 
   describe('update', function() {
-    let contactId
+    let contactId = 123
+    const updateData = {
+      properties: [
+        {
+          property: 'email',
+          value: `new-email${Date.now()}@hubspot.com`,
+        },
+        {
+          property: 'firstname',
+          value: 'Updated',
+          timestamp: Date.now(),
+        },
+        {
+          property: 'lastname',
+          value: 'Lead',
+          timestamp: Date.now(),
```

Comment:
Just curious, is the `timestamp` necessary? I'm also interested in the decision to update these particular fields. Can this test update fewer fields and still confirm `update` works as expected?

Reply:
I removed them and it seems fine 
https://github.com/MadKudu/node-hubspot/pull/140/commits/2026e197fbbbb2b6e152cf10f74af9559ceaa224